### DISCUSSION TITLE
Significant increase in input latency when using high polling rate mouse

### DISCUSSION THREAD

**Discussion by mangkoran** [ppy/osu#21593] [Q&A] [answered]
### Type

Performance

### Bug description

I noticed increased input latency when I am using high polling rate setting (1000 Hz) on my mouse (Fantech Aria, Delux M800DB). Lowering polling rate will make the input latency better (125>250>500>1000).

This issue has been raised several times before ([1](https://github.com/ppy/osu/issues/10780), [2](https://github.com/ppy/osu/issues/11913)), but I would like to raise it once again as I could not understand why this is caused by external program other than osu! as in Windows itself I noticed no significant input latency or delay.

I attached video with Task Manager below showing top tasks that use most CPU during the gameplay recording.

### Screenshots or videos

https://user-images.githubusercontent.com/57625126/206761918-8532c966-d06e-41bc-8309-c6405fd01ddd.mp4

https://user-images.githubusercontent.com/57625126/206761936-889c1f7b-f8f0-4ba5-bdc2-cf773f03434a.mp4

### Version

2022.1208.0-lazer

### Logs

[database.log](https://github.com/ppy/osu/files/10196935/database.log)
[input.log](https://github.com/ppy/osu/files/10196936/input.log)
[network.log](https://github.com/ppy/osu/files/10196937/network.log)
[performance.log](https://github.com/ppy/osu/files/10196938/performance.log)
[runtime.log](https://github.com/ppy/osu/files/10196939/runtime.log)
[updater.log](https://github.com/ppy/osu/files/10196940/updater.log)

**Comment by peppy** [comment, marked as answer]
Performance and optimisation considerations are an ongoing effort.

What you are seeing here is the game's overhead from processing so many inputs per frame (it's not cheap). This will improve as we continue to optimise the game, but it will never be zero, so it's a balancing act.

One thing to note is that if it's another app on your system causing overhead, it will show up in the input thread as grey/white area. In your case, you can see an increase, but it's not enough to matter.

**Reply by mangkoran** [reply]
Alright. Now I know that this is known issue/will be  looked later on. Thanks peppy!
